clients/privacy-center/app/server-environment.ts

```diff
@@ -250,6 +254,9 @@ export const loadPrivacyCenterEnvironment =
       FIDES_API_URL:
         process.env.FIDES_PRIVACY_CENTER__FIDES_API_URL ||
         "http://localhost:8080/api/v1",
+      SERVER_SIDE_FIDES_API_URL:
+        process.env.FIDES_PRIVACY_CENTER__SERVER_SIDE_FIDES_API_URL ||
+        "http://fides:8080/api/v1",
```

Comment:
🤔 

Setting a really smart default value here will save us a lot of headaches. Right now the default will be `http://fides:8080/api/v1`, which will work in a local, `fides deploy`-esque setup.

However, the engineer deploying this to a hosted environment will need to set both `FIDES_PRIVACY_CENTER__FIDES_API_URL` and `FIDES_PRIVACY_CENTER__SERVER_SIDE_API_URL` to make it work there.

My instinct says you should do this instead:
```suggestion
      SERVER_SIDE_FIDES_API_URL:
        process.env.FIDES_PRIVACY_CENTER__SERVER_SIDE_FIDES_API_URL ||
        null
```

...and wherever this variable is used, do something like: `let apiUl = settings.SERVER_SIDE_FIDES_API_URL || settings.FIDES_API_URL`. In other words, I'd recommend we make this an *optional* variable that, when set, will be used for server-side fetches. But if it's not set, we use the value of the `FIDES_API_URL` config by default.

To make that work and still support local testing, you'll need to manually override this value on the docker-compose files in this repo, the sample app, and the demo repo... but you'd need to do that anyways to control `IS_PREFETCH_ENABLED` too!

The benefit here is:
* for the production use case, the engineer deploying this sets one variable (FIDES_API_URL) and it'll work
* in special cases, the engineer can use this variable to have different API URLs for client-side and server-side fetches
* we will need to use that "special case" for local deployments where the API URLs are different

Reply:
ahh so smart. Such a good idea to fall back on `FIDES_API_URL` if the `SERVER_SIDE_FIDES_API_URL` is not provided. Then it's also clearer that it is a special case when you have to set 2 different API urls 🌟